holoviews/plotting/bokeh/element.py

```diff
@@ -1906,6 +1914,36 @@ def initialize_plot(self, ranges=None, plot=None, plots=None, source=None):
 
         return plot
 
+    def _apply_hard_bound(self, element, ranges):
+        """
+        Apply hard bounds to the x and y ranges of the plot.
+
+        Sets the navigable bounds of the plot based on the extents
+        of the given element and ranges. If an extend is numeric and not NaN, it is
+        used as is. Otherwise, it is set to None, which means that end of the axis
+        is unbounded.
+        """
+
+        # Skip if the element doesn't have an 'extents' attribute
+        if not hasattr(element, 'extents'):
```

Comment:
`extents` is soft-deprecated so making anything conditional on it is weird. That said I think most elements do have it.

Reply:
We spoke last week and said that perhaps defaulting to True might be ok, as long as it defaults to False if projection is set. So after that, I used:

```python
# If apply_hard_bound is not set but projection has been set, avoid applying hard bounds by default
# since a projection typically means that the plot is either geographic, polar, or 3d
if self.apply_hard_bounds or (self.projection is None and self.apply_hard_bounds is None):
    self._apply_hard_bounds(element, ranges)
```

But I've since determined that it's best to keep hard bounds default as false so I've simplified

```python
if self.apply_hard_bounds:
    self._apply_hard_bounds(element, ranges)
```